app/scripts/lib/util.js

```diff
@@ -173,13 +162,47 @@ function isPrefixedFormattedHexString(value) {
   return /^0x[1-9a-f]+[0-9a-f]*$/iu.test(value)
 }
 
+/**
+ * Returns string with '0x' hex prefix
+ *
+ * @param {string} str
+ * @returns {string}
+ */
+const addHexPrefix = (str) => {
+  if (
+    typeof str !== 'string' ||
+    str.toLowerCase().startsWith('-0x') ||
+    str.toLowerCase().startsWith('0x')
+  ) {
+    return typeof str === 'string' ? str.replace('0X', '0x') : str
+  }
```

Comment:
For the sake of clarity and specificity:
```suggestion
  if (typeof str !== 'string') {
    return str
  }

  if (str.match(/^-?0X/u)) {
    return str.replace(/0X/u, '0x')
  }
```